cmake/SerialboxConfig.cmake.in

```diff
@@ -40,12 +40,78 @@ set(SERIALBOX_BOOST_LIBRARY_DIRS "@Boost_LIBRARY_DIRS@")
 set(SERIALBOX_REQUIRED_BOOST_COMPONENTS "@REQUIRED_BOOST_COMPONENTS@")
 
+#===---------------------------------------------------------------------------------------------===
+#   Helper functions used in this file
+#====--------------------------------------------------------------------------------------------===
+
+function( serialbox_print_targets )
+  #
+  # serialbox_print_targets( targets_list )
+  #
+  #    Print exported and unavailable targets
+  #
+  message( STATUS "Serialbox exported targets:")
+  foreach( target ${ARGV} )
+    if( TARGET ${target} )
+      message( STATUS "  ${target}" )
+    else()
+      list( APPEND _unavailable_targets ${target} )
+    endif()
+  endforeach()
+  if( _unavailable_targets )
+    message( STATUS "Serialbox unavailable targets:")
+    foreach( target ${_unavailable_targets} )
+      message( STATUS "  ${target}" )
+    endforeach()
+  endif()
+endfunction()
+
+function( serialbox_alias_targets )
+  #
+  # serialbox_alias_targets( alias target_shared target_static align )
+  #
+  #    Creates ALIAS target for given targets. Prefer target_shared if availabe
+  # 
+  set( _alias ${ARGV0} )
+  set( _target_shared ${ARGV1} )
+  set( _target_static ${ARGV2} )
+  set( _align ${ARGV3} )
+  string( REPLACE "Serialbox::" "" _target_shared_build "${_target_shared}" )
+  string( REPLACE "Serialbox::" "" _target_static_build "${_target_static}" )
+  if( TARGET ${_target_shared_build} )
+    message( STATUS "  ${_alias} ${_align}--> Serialbox::${_target_shared_build}" )
+    if( NOT TARGET ${_alias} )
+      add_library( ${_alias} ALIAS ${_target_shared_build} )
+    endif()
+  elseif( TARGET ${_target_shared} AND CMAKE_VERSION GREATER 3.11 )
+    message( STATUS "  ${_alias} ${_align}--> ${_target_shared}" )
+    set_target_properties( ${_target_shared} PROPERTIES IMPORTED_GLOBAL TRUE) # required for aliasing imports
+    if( NOT TARGET ${_alias} )
+      add_library( ${_alias} ALIAS ${_target_shared} )
+    endif()
+  elseif( TARGET ${_target_static_build} )
+    message( STATUS "  ${_alias} ${_align}--> Serialbox::${_target_static_build}" )
+    if( NOT TARGET ${_alias} )
+      add_library( ${_alias} ALIAS ${_target_static_build} )
+    endif()
+  elseif( TARGET ${_target_static} AND CMAKE_VERSION GREATER 3.11 )
+    message( STATUS "  ${_alias} ${_align}--> ${_target_static}" )
+    set_target_properties( ${_target_static} PROPERTIES IMPORTED_GLOBAL TRUE) # required for aliasing imports
+    if( NOT TARGET ${_alias} )
+      add_library( ${_alias} ALIAS ${_target_static} )
+    endif()
+  endif()
+endfunction()
+
 #===---------------------------------------------------------------------------------------------===
 #   Find Serialbox libraries
 #====--------------------------------------------------------------------------------------------===
 
 # Import library targets 
-include("${CMAKE_CURRENT_LIST_DIR}/SerialboxTargets.cmake")
+if( NOT @PROJECT_NAME@_TARGETS_EXPORTED )
```

Comment:
Why is this protection necessary?

Reply:
If you are in a nested CMake tree like:
```
super-cmake-project/
    serialbox2/
    myproject/
```
and `myproject` has a `find_package(SerialBox)` then the targets are already globally visible, so we must avoid including the targets again. the `@PROJECT_NAME@_TARGETS_EXPORTED` variable is set during the configuration of `serialbox2`